### DISCUSSION TITLE
Pinia 2.2.2 not working with Vue 3.4.38

### DISCUSSION THREAD

**Discussion by GorillaWarHead** [vuejs/pinia#2752] [Help and Questions] [answered]
### Reproduction

its in install problem 

### Steps to reproduce the bug

in a new nuxt project install pinia

npm i pinia 

### Expected behavior

expected pinia to be installed

### Actual behavior

While resolving: nuxt-app@undefined
Found: vue@3.4.38
node_modules/vue
  peer vue@"^3.0.0-0 || ^2.6.0" from vue-demi@0.14.10
  node_modules/@floating-ui/vue/node_modules/vue-demi
    vue-demi@">=0.13.0" from @floating-ui/vue@1.1.4
    node_modules/@floating-ui/vue
      @floating-ui/vue@"^1.1.0" from radix-vue@1.9.5
      node_modules/radix-vue
        radix-vue@"^1.4.9" from vaul-vue@0.2.0
        node_modules/vaul-vue
          vaul-vue@"^0.2.0" from the root project
        radix-vue@"^1.9.2" from the root project
  peer vue@">=3" from @iconify/vue@4.1.2
  node_modules/@iconify/vue
    dev @iconify/vue@"^4.1.2" from the root project
  peer vue@"^3.3.4" from @nuxt/vite-builder@3.13.0
  node_modules/@nuxt/vite-builder
    @nuxt/vite-builder@"3.13.0" from nuxt@3.13.0
    node_modules/nuxt
      dev nuxt@"^3.13.0" from the root project
  peer vue@"^2.6.14 || ^3.3.0" from pinia@2.2.2
  node_modules/@pinia/nuxt/node_modules/pinia
    pinia@"^2.2.2" from @pinia/nuxt@0.5.4
    node_modules/@pinia/nuxt
      @pinia/nuxt@"^0.5.4" from the root project
      peer @pinia/nuxt@"^0.5.0" from @pinia-plugin-persistedstate/nuxt@1.2.1
      node_modules/@pinia-plugin-persistedstate/nuxt
        dev @pinia-plugin-persistedstate/nuxt@"^1.2.1" from the root project
  peer vue@"^3.0.0-0 || ^2.6.0" from vue-demi@0.14.10
  node_modules/@pinia/nuxt/node_modules/vue-demi
    vue-demi@"^0.14.10" from pinia@2.2.2
    node_modules/@pinia/nuxt/node_modules/pinia
      pinia@"^2.2.2" from @pinia/nuxt@0.5.4
      node_modules/@pinia/nuxt
        @pinia/nuxt@"^0.5.4" from the root project
        peer @pinia/nuxt@"^0.5.0" from @pinia-plugin-persistedstate/nuxt@1.2.1
        node_modules/@pinia-plugin-persistedstate/nuxt
          dev @pinia-plugin-persistedstate/nuxt@"^1.2.1" from the root project
  peer vue@">=3.2" from @tanstack/vue-table@8.20.5
  node_modules/@tanstack/vue-table
    @tanstack/vue-table@"^8.19.3" from the root project
  peer vue@"^2.7.0 || ^3.0.0" from @tanstack/vue-virtual@3.10.4
  node_modules/@tanstack/vue-virtual
    @tanstack/vue-virtual@"^3.8.1" from radix-vue@1.9.5
    node_modules/radix-vue
      radix-vue@"^1.4.9" from vaul-vue@0.2.0
      node_modules/vaul-vue
        vaul-vue@"^0.2.0" from the root project
      radix-vue@"^1.9.2" from the root project
  peer vue@">=2.7 || >=3" from @unhead/vue@1.10.0
  node_modules/@unhead/vue
    @unhead/vue@"^1.10.0" from nuxt@3.13.0
    node_modules/nuxt
      dev nuxt@"^3.13.0" from the root project
  peer vue@"^3.2.25" from @vitejs/plugin-vue@5.1.2
  node_modules/@vitejs/plugin-vue
    @vitejs/plugin-vue@"^5.1.2" from @nuxt/vite-builder@3.13.0
    node_modules/@nuxt/vite-builder
      @nuxt/vite-builder@"3.13.0" from nuxt@3.13.0
      node_modules/nuxt
        dev nuxt@"^3.13.0" from the root project
  peer vue@"^3.0.0" from @vitejs/plugin-vue-jsx@4.0.1
  node_modules/@vitejs/plugin-vue-jsx
    @vitejs/plugin-vue-jsx@"^4.0.1" from @nuxt/vite-builder@3.13.0
    node_modules/@nuxt/vite-builder
      @nuxt/vite-builder@"3.13.0" from nuxt@3.13.0
      node_modules/nuxt
        dev nuxt@"^3.13.0" from the root project
  peerOptional vue@"^2.7.0 || ^3.2.25" from @vue-macros/common@1.12.2
  node_modules/@vue-macros/common
    @vue-macros/common@"^1.12.2" from unplugin-vue-router@0.10.7
    node_modules/unplugin-vue-router
      unplugin-vue-router@"^0.10.7" from nuxt@3.13.0
      node_modules/nuxt
        dev nuxt@"^3.13.0" from the root project
  peer vue@"3.4.38" from @vue/server-renderer@3.4.38
  node_modules/@vue/server-renderer
    @vue/server-renderer@"3.4.38" from vue@3.4.38
  peer vue@"^3.0.0-0 || ^2.6.0" from vue-demi@0.14.10
  node_modules/@vueuse/core/node_modules/vue-demi
    vue-demi@">=0.14.8" from @vueuse/core@10.11.1
    node_modules/@vueuse/core
      @vueuse/core@"^10.11.0" from radix-vue@1.9.5
      node_modules/radix-vue
        radix-vue@"^1.4.9" from vaul-vue@0.2.0
        node_modules/vaul-vue
          vaul-vue@"^0.2.0" from the root project
        radix-vue@"^1.9.2" from the root project
      @vueuse/core@"^10.8.0" from vaul-vue@0.2.0
      node_modules/vaul-vue
        vaul-vue@"^0.2.0" from the root project
      @vueuse/core@"^10.11.0" from the root project
  peer vue@"^3.0.0-0 || ^2.6.0" from vue-demi@0.14.10
  node_modules/@vueuse/shared/node_modules/vue-demi
    vue-demi@">=0.14.8" from @vueuse/shared@10.11.1
    node_modules/@vueuse/shared
      @vueuse/shared@"10.11.1" from @vueuse/core@10.11.1
      node_modules/@vueuse/core
        @vueuse/core@"^10.11.0" from radix-vue@1.9.5
        node_modules/radix-vue
          radix-vue@"^1.4.9" from vaul-vue@0.2.0
          node_modules/vaul-vue
            vaul-vue@"^0.2.0" from the root project
          radix-vue@"^1.9.2" from the root project
        @vueuse/core@"^10.8.0" from vaul-vue@0.2.0
        node_modules/vaul-vue
          vaul-vue@"^0.2.0" from the root project
        @vueuse/core@"^10.11.0" from the root project
      @vueuse/shared@"^10.11.0" from radix-vue@1.9.5
      node_modules/radix-vue
        radix-vue@"^1.4.9" from vaul-vue@0.2.0
        node_modules/vaul-vue
          vaul-vue@"^0.2.0" from the root project
        radix-vue@"^1.9.2" from the root project
  peer vue@">=3.0.1" from lucide-vue-next@0.416.0
  node_modules/lucide-vue-next
    lucide-vue-next@"^0.416.0" from the root project
  vue@"^3.4.38" from nuxt@3.13.0
  node_modules/nuxt
    dev nuxt@"^3.13.0" from the root project
  peer vue@">= 3.2.0" from radix-vue@1.9.5
  node_modules/radix-vue
    radix-vue@"^1.4.9" from vaul-vue@0.2.0
    node_modules/vaul-vue
      vaul-vue@"^0.2.0" from the root project
    radix-vue@"^1.9.2" from the root project
  vue@"^3.4.5" from vaul-vue@0.2.0
  node_modules/vaul-vue
    vaul-vue@"^0.2.0" from the root project
  peer vue@"^3.0.0" from vue-i18n@9.14.0
  node_modules/vue-i18n
    peerOptional vue-i18n@"*" from @intlify/unplugin-vue-i18n@3.0.1
    node_modules/@intlify/unplugin-vue-i18n
      @intlify/unplugin-vue-i18n@"^3.0.1" from @nuxtjs/i18n@8.5.1
      node_modules/@nuxtjs/i18n
        @nuxtjs/i18n@"^8.3.3" from the root project
    vue-i18n@"^9.9.0" from @nuxtjs/i18n@8.5.1
    node_modules/@nuxtjs/i18n
      @nuxtjs/i18n@"^8.3.3" from the root project
  peer vue@"^3.2.0" from vue-router@4.4.3
  node_modules/vue-router
    vue-router@"^4.4.0" from @nuxtjs/i18n@8.5.1
    node_modules/@nuxtjs/i18n
      @nuxtjs/i18n@"^8.3.3" from the root project
    vue-router@"^4.4.3" from nuxt@3.13.0
    node_modules/nuxt
      dev nuxt@"^3.13.0" from the root project
    peerOptional vue-router@"^4.4.0" from unplugin-vue-router@0.10.7
    node_modules/unplugin-vue-router
      unplugin-vue-router@"^0.10.7" from nuxt@3.13.0
      node_modules/nuxt
        dev nuxt@"^3.13.0" from the root project
  vue@"latest" from the root project

Could not resolve dependency:
pinia@"^2.2.2" from the root project

Conflicting peer dependency: vue@2.6.14
node_modules/vue
  peer vue@">= 2.5 < 2.7" from @vue/composition-api@1.7.2
  node_modules/@vue/composition-api
    peerOptional @vue/composition-api@"^1.4.0" from pinia@2.2.2
    node_modules/pinia
      pinia@"^2.2.2" from the root project

Fix the upstream dependency conflict, or retry
this command with --force or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.

### Additional information

Node.js v20.17.0

my package.json

```json
{
  "name": "nuxt-app",
  "private": true,
  "type": "module",
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev --host",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "dependencies": {
    "@nuxtjs/i18n": "^8.3.3",
    "@pinia/nuxt": "^0.5.4",
    "@tanstack/vue-table": "^8.19.3",
    "@vueuse/core": "^10.11.0",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.1.1",
    "lucide-vue-next": "^0.416.0",
    "pinia": "^2.2.2",
    "radix-vue": "^1.9.2",
    "shadcn-nuxt": "^0.10.4",
    "tailwind-merge": "^2.4.0",
    "tailwindcss-animate": "^1.0.7",
    "vaul-vue": "^0.2.0",
    "vue": "latest"
  },
  "devDependencies": {
    "@iconify/vue": "^4.1.2",
    "@nuxtjs/color-mode": "^3.4.2",
    "@nuxtjs/tailwindcss": "^6.12.1",
    "@pinia-plugin-persistedstate/nuxt": "^1.2.1",
    "nuxt": "^3.13.0",
    "typescript": "^5.5.4"
  }
}

```

**Comment by gsabater** [comment]
I had this problem recently and i slapped the error to chatgpt. it suggested me to specify @vue/composition-api to false in the overrides. Haven't had any errors so far... so that's it 🤷‍♀️

In packjage.json
```
"overrides": {
    "vue": "^3.5.3",
    "@vue/composition-api": false
  },
```

**Reply by shery** [reply]
worked for me

**Comment by kaiser54** [comment]
[https://github.com/vuejs/pinia/discussions/2397#discussioncomment-10642772](url)

---

All I did was

1. **Install Vue**:
   ```bash
   npm install vue
   ```

2. **Install Pinia**:
   ```bash
   npm install pinia @pinia/nuxt
   ```

3. **Add Pinia module to `nuxt.config.ts`**:
   ```js
   export default defineNuxtConfig({
     modules: ['@pinia/nuxt'],
   });
   ```

4. **Import `defineStore` from Pinia**:
   ```js
   import { defineStore } from 'pinia';
   ```

This setup worked perfectly for me! 👌🏽

**Comment by galaxyblur** [comment, marked as answer]
I think the Nuxt project generator gave me `"vue": "latest"` in my `package.json`. Switching this to `"vue": "^3.5.10"` fixed the issue for me.

**Reply by kaiser54** [reply]
yeah
